File: 1464. Maximum Product of Two Elements in an Array/solution.java
class Solution {
  public int maxProduct(int[] nums) {
    int firstMax = 0;
    int secondMax = 0;
    for (int index = 0; index < nums.length; index++){
      int num = nums[index];
      if (num >= firstMax){
        secondMax = firstMax;
        firstMax = num;
        continue;
      }
      if (num > secondMax){
        secondMax = num;
      }
    }

    return (firstMax - 1) * (secondMax - 1);
  }
}

File: 1464. Maximum Product of Two Elements in an Array/solution.py
class Solution(object):
  def maxProduct(self, nums):
    """
    :type nums: List[int]
    :rtype: int
    """
    firstMax = 0
    secondMax = 0
    for num in nums:
      if num >= firstMax:
        secondMax = firstMax
        firstMax = num
        continue
      if num > secondMax:
        secondMax = num
        
    return (firstMax - 1) * (secondMax - 1)